pkg/iac/rego/load.go

```diff
@@ -353,12 +353,14 @@ func (s *Scanner) filterModules() error {
 		if metadata != nil {
 			tv, err := semver.Parse(s.trivyVersion)
 			if err != nil {
-				s.logger.Warn(
-					"Failed to parse Trivy version - cannot confirm if module will work with current version",
-					log.String("trivy_version", s.trivyVersion),
-					log.FilePath(module.Package.Location.File),
-					log.Err(err),
-				)
+				if s.trivyVersion != "dev" {
+					s.logger.Warn(
```

Comment:
I think we can have another logger method such as `.WarnIf()` where we could conditionally log messages based on a certain boolean condition.